test/functional/vimscript/system_spec.lua

```diff
@@ -173,6 +173,23 @@ describe('system()', function()
         eq('C:\\\n', eval([[system('cd c:\; (Get-Location).Path')]]))
         eq('a b\n', eval([[system('Write-Output "a b"')]]))
       end)
+
+      it('works with powershell for filter and redirection (#16271)', function()
+        local tempfile = helpers.tmpname()
+        helpers.set_shell_powershell()
+        screen:try_resize(500, 8)
+        feed(':edit ' .. tempfile .. '<cr>')
+        feed('i3<cr>1<cr>2<esc>:write<cr>')
+        feed(':4verbose %!sort<cr>')
+        screen:expect{
+          any=[[3 lines filtered]]
+        }
+        screen:expect{
+          unchanged=true,
+          any=[[Executing command: "Start%-Process sort %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
+        }
```

Comment:
should be able to avoid the `unchanged=true` by combining these:

```suggestion
        screen:expect{
          any=[[3 lines filtered.*Executing command: "Start%-Process sort %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
        }
```

Reply:
Done.